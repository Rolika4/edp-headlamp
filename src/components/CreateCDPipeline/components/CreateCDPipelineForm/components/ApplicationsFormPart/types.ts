import { SyntheticEvent } from 'react';
import { FormNameObject } from '../../../../../../types/forms';

export interface ApplicationsFormPartProps {
    names: {
        [key: string]: FormNameObject;
    };
    handleFormFieldChange(event: SyntheticEvent): void;
}
