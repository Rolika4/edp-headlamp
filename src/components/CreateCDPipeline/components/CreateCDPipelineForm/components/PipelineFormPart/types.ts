import { SyntheticEvent } from 'react';
import { FormNameObject } from '../../../../../../types/forms';

export interface PipelineInfoFormPartProps {
    names: {
        [key: string]: FormNameObject;
    };
    handleFormFieldChange(event: SyntheticEvent): void;
}
