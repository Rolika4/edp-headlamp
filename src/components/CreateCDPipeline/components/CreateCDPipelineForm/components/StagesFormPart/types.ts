import { EDPCDPipelineStageKubeObjectInterface } from '../../../../../../k8s/EDPCDPipelineStage/types';
import { React } from '../../../../../../plugin.globals';
import { DeepPartial } from '../../../../../../types/global';

export interface StagesFormPartProps {
    setCreateStageDialogOpen: React.Dispatch<React.SetStateAction<boolean>>;
    stages: DeepPartial<EDPCDPipelineStageKubeObjectInterface>;
    onStageDelete: (idx: number) => void;
}
