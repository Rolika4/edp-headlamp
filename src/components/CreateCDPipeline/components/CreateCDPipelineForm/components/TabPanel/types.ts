import { React } from '../../../../../../plugin.globals';

export interface TabPanelProps extends React {
    children?: React.ReactNode;
    index: any;
    value: any;
    className: string;
}
