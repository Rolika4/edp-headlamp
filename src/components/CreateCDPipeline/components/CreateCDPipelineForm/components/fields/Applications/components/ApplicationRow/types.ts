import { SyntheticEvent } from 'react';
import { FormNameObject } from '../../../../../../../../../types/forms';
import { Application } from '../../types';

export interface ApplicationRowProps {
    names: { [key: string]: FormNameObject };
    application: Application;
    setApplications(prev: any): Application[];
    handleFormFieldChange(event: SyntheticEvent): void;
}
