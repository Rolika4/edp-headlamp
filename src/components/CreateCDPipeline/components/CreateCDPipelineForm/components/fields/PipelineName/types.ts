import { CreationFormFieldInterface } from '../../../../../../../types/forms';

export interface PipelineNameProps extends CreationFormFieldInterface {}
