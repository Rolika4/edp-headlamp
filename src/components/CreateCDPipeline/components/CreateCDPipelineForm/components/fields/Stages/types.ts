import { EDPCDPipelineStageKubeObjectInterface } from '../../../../../../../k8s/EDPCDPipelineStage/types';
import { DeepPartial } from '../../../../../../../types/global';

export interface StagesProps {
    stages: DeepPartial<EDPCDPipelineStageKubeObjectInterface>;
    setCreateStageDialogOpen: () => void;
    onStageDelete: (idx: number) => void;
}
