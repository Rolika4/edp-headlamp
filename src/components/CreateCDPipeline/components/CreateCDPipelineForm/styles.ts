import { MuiCore } from '../../../../plugin.globals';
import { rem } from '../../../../utils/styling/rem';

const { makeStyles } = MuiCore;

export const useStyles = makeStyles((theme: DefaultTheme) => ({
    form: {
        width: '80%',
        position: 'relative',
        paddingRight: rem(16),

        '& form': {
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'space-between',
            height: '100%',
        },
    },
    formInner: {
        flex: `1 0 calc(100% - ${rem(75)})`,
        minHeight: rem(100),
        overflowY: 'auto',
    },
    tabs: {
        backgroundColor: theme.palette.background.default,
        borderRadius: rem(5),
        width: '20%',

        '& .MuiTab-wrapper': {
            alignItems: 'flex-start',
        },

        '& .MuiTab-root': {
            padding: `${rem(6)} ${rem(12)} ${rem(6)} ${rem(20)}`,
        },
    },
    tabPanel: {
        height: '100%',
        display: 'flex',
        flexDirection: 'column',
        position: 'relative',

        '&[hidden]': {
            height: 0,
            overflow: 'hidden',
        },
    },
    tabPanelInner: {
        padding: `${rem(16)} ${rem(16)} ${rem(32)} ${rem(16)}`,
    },
    tabPanelActions: {
        flexGrow: 1,
        display: 'flex',
        alignItems: 'center',
        gap: rem(20),
        padding: rem(20),
        backgroundColor: theme.palette.background.paper,
        boxShadow: `0 ${rem(-5)} ${rem(54)} ${rem(-28)} rgba(0,0,0, 0.38)`,
    },
}));
