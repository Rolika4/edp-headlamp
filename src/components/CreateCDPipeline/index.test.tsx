/**
 * @jest-environment jsdom
 */

import { describe, jest } from '@jest/globals';
import { createTheme } from '@material-ui/core';
import { ThemeProvider } from '@material-ui/styles';
import { configureStore } from '@reduxjs/toolkit';
import { render, screen, waitFor } from '@testing-library/react';
import { SnackbarProvider } from 'notistack';
import React from 'react';
import { Provider } from 'react-redux';
import { namespacesMock } from '../../hooks/useNamespaces/mocks/namespaces.mock';
import { pluginLib } from '../../plugin.globals';
import { CreateCDPipeline } from './index';
import { CreateCDPipelineProps } from './types';

const { ApiProxy } = pluginLib;

describe('CreateCDPipeline', () => {
    it('should render correctly', async () => {
        const props: CreateCDPipelineProps = {
            createDialogOpen: true,
            setCreateDialogOpen: () => {},
            onClose: () => {},
        };

        const store = configureStore({
            reducer: () => ({}),
        });

        const theme: DefaultTheme = createTheme();

        jest.spyOn(ApiProxy, 'request').mockImplementation(url => {
            if (url.includes('/api/v1/namespaces')) {
                return Promise.resolve(namespacesMock);
            }
        });

        render(
            <Provider store={store}>
                <SnackbarProvider>
                    <ThemeProvider theme={theme}>
                        <CreateCDPipeline {...props} />
                    </ThemeProvider>
                </SnackbarProvider>
            </Provider>
        );

        await waitFor(() => {
            expect(screen.getByTestId('create-cdpipeline')).toMatchSnapshot();
        });
    });
});
