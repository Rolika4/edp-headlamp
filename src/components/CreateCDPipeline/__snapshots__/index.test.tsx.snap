// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateCDPipeline should render correctly 1`] = `
<div
  class="makeStyles-dialog-2"
  data-testid="create-cdpipeline"
>
  <div
    class="makeStyles-dialogTitle-3"
  >
    <h5
      class="MuiTypography-root MuiTypography-h5"
    >
      Create CD Pipeline
    </h5>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeSmall"
        >
          <span />
        </span>
        Edit YAML
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div
    class="MuiDialogContent-root makeStyles-dialogContent-4"
  >
    <div
      class="MuiTabs-root makeStyles-tabs-7 MuiTabs-vertical"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed"
        style="overflow: hidden;"
      >
        <div
          aria-label="simple tabs example"
          class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
          role="tablist"
        >
          <button
            aria-controls="simple-tabpanel-0"
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected"
            id="simple-tab-0"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Pipeline
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-1"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
            id="simple-tab-1"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Applications
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            aria-controls="simple-tabpanel-2"
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary"
            id="simple-tab-2"
            role="tab"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiTab-wrapper"
            >
              Stages
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="PrivateTabIndicator-root-11 PrivateTabIndicator-colorPrimary-12 MuiTabs-indicator PrivateTabIndicator-vertical-14"
          style="top: 0px; height: 0px;"
        />
      </div>
    </div>
    <div
      class="makeStyles-form-5"
    >
      <form>
        <div
          class="makeStyles-formInner-6"
        >
          <div
            aria-labelledby="simple-tab-0"
            class="makeStyles-tabPanel-8"
            id="simple-tabpanel-0"
            role="tabpanel"
          >
            <div
              class="makeStyles-tabPanelInner-9"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-15"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-16 MuiTypography-body1"
                            >
                              Pipeline name
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex; align-items: flex-end;"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiFormControl-root MuiTextField-root"
                            >
                              <div
                                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                              >
                                <input
                                  aria-invalid="false"
                                  class="MuiInputBase-input MuiInput-input"
                                  name="name"
                                  placeholder="Enter pipeline name"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                      style="display: flex;"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <span
                            class="makeStyles-labelWrap-15"
                          >
                            <span
                              class="MuiTypography-root makeStyles-label-16 MuiTypography-body1"
                            >
                              Deployment type
                            </span>
                          </span>
                        </div>
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                        >
                          <div
                            class="MuiFormControl-root MuiFormControl-fullWidth"
                          >
                            <div
                              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                            >
                              <div
                                aria-haspopup="listbox"
                                aria-labelledby="mui-component-select-deploymentType"
                                class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                                id="mui-component-select-deploymentType"
                                role="button"
                                tabindex="0"
                              >
                                Container
                              </div>
                              <input
                                aria-hidden="true"
                                class="MuiSelect-nativeInput"
                                name="deploymentType"
                                tabindex="-1"
                                value="container"
                              />
                              <svg
                                aria-hidden="true"
                                class="MuiSvgIcon-root MuiSelect-icon"
                                focusable="false"
                                viewBox="0 0 24 24"
                              >
                                <path
                                  d="M7 10l5 5 5-5z"
                                />
                              </svg>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-1"
            class="makeStyles-tabPanel-8"
            hidden=""
            id="simple-tabpanel-1"
            role="tabpanel"
          >
            <div
              class="makeStyles-tabPanelInner-9"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                      >
                        <div
                          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                          style="display: flex;"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            >
                              <span
                                class="makeStyles-labelWrap-15"
                              >
                                <span
                                  class="MuiTypography-root makeStyles-label-16 MuiTypography-body1"
                                >
                                  Applications
                                </span>
                              </span>
                            </div>
                            <div
                              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                            >
                              <div
                                class="MuiFormControl-root MuiFormControl-fullWidth"
                              >
                                <div
                                  class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-selectWithDefaultValue-17 Mui-disabled Mui-disabled MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
                                >
                                  <div
                                    aria-disabled="true"
                                    aria-haspopup="listbox"
                                    aria-labelledby="mui-component-select-applicationsToAddChooser"
                                    class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input Mui-disabled Mui-disabled Mui-disabled"
                                    id="mui-component-select-applicationsToAddChooser"
                                    role="button"
                                  >
                                    Select applications
                                  </div>
                                  <input
                                    aria-hidden="true"
                                    class="MuiSelect-nativeInput"
                                    name="applicationsToAddChooser"
                                    tabindex="-1"
                                    value=""
                                  />
                                  <svg
                                    aria-hidden="true"
                                    class="MuiSvgIcon-root MuiSelect-icon Mui-disabled"
                                    focusable="false"
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M7 10l5 5 5-5z"
                                    />
                                  </svg>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
                      style="display: flex; align-items: center; justify-content: flex-end; flex-direction: column;"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
                        disabled=""
                        style="min-width: 0;"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          add
                        </span>
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
                  />
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation2"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiAlert-message"
                    >
                      Add at least one application
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            aria-labelledby="simple-tab-2"
            class="makeStyles-tabPanel-8"
            hidden=""
            id="simple-tabpanel-2"
            role="tabpanel"
          >
            <div
              class="makeStyles-tabPanelInner-9"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
              >
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-h6"
                      >
                        Stages
                      </h6>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <div
                        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
                      />
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSizeSmall MuiButton-sizeSmall"
                        style="min-width: 0;"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiButton-label"
                        >
                          add stage
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
                >
                  <div
                    class="MuiPaper-root MuiAlert-root MuiAlert-filledInfo MuiPaper-elevation2"
                    role="alert"
                  >
                    <div
                      class="MuiAlert-icon"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
                        />
                      </svg>
                    </div>
                    <div
                      class="MuiAlert-message"
                    >
                      Add at least one stage
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="makeStyles-tabPanelActions-10"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              undo changes
            </span>
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
            style="margin-left: auto;"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              cancel
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              proceed
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
